app/code/Magento/InventoryCatalogAdminUi/Plugin/InventoryApi/MaintainSourceItemStatusConsistencyPlugin.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogAdminUi\Plugin\InventoryApi;
+
+use Magento\InventoryApi\Api\SourceItemsSaveInterface;
+
+/**
+ * Maintains consistency with Single-Source mode in case when no Source Item Quantity provided.
+ */
+class MaintainSourceItemStatusConsistencyPlugin
+{
+    /**
+     * Set Status to 0 if no Quantity provided.
+     *
+     * @param SourceItemsSaveInterface $subject
+     * @param \Magento\InventoryApi\Api\Data\SourceItemInterface[] $sourceItems
+     * @return array
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeExecute(SourceItemsSaveInterface $subject, array $sourceItems): array
+    {
+        foreach ($sourceItems as $sourceItem) {
+            if ('' === $sourceItem->getQuantity()) {
+                $sourceItem->setStatus(0);
```

Comment:
Why do we change stock status automatically?
We considered that it is the normal situation to have qty=0 and status in_stock=1 (because it is only conditions for is_salable fields)